File: python-examples/native-crawler/utils/sanitize_key.py
def sanitize_key(key: str) -> str:
    for char in ["://", "/", ":", "#", "?", "&", "=", ".", "-"]:
        key = key.replace(char, "_")
    while "__" in key:
        key = key.replace("__", "_")
    return key.strip("_")

File: typescript-examples/native-crawler/utils/sanitize_key.ts

export function sanitizeKey(key: string): string {
  const chars = ["://", "/", ":", "#", "?", "&", "=", ".", "-"];
  let sanitized = key;

  for (const char of chars) {
    sanitized = sanitized.replaceAll(char, "_");
  }

  while (sanitized.includes("__")) {
    sanitized = sanitized.replace("__", "_");
  }

  return sanitized.replace(/^_+|_+$/g, "");
}
